components/fee-button.js

```diff
@@ -36,8 +36,7 @@ export function postCommentBaseLineItems ({ baseCost = 1, comment = false, me })
   }
 }
 
-export function postCommentUseRemoteLineItems ({ parentId, me } = {}) {
-  if (!me) return () => {}
```

Comment:
I hope this doesn't break anything but afaict, it doesn't. It just does unnecessary requests since anon has fixed fees.

Reply:
This isn't a hook though. It's a function that returns a hook

Reply:
> It's a function that returns a hook

Yes but conditionally

Reply:
> but I believe this is already conditional

Yes, it is. That's what my change is trying to fix since the bug would manifest when switching to anon. The user report shows that the bug _already_ manifests in some cases.  

Example:

This is the full function:

```js
export function postCommentUseRemoteLineItems ({ parentId, me } = {}) {
  if (!me) return () => {}
  const query = parentId
    ? gql`{ itemRepetition(parentId: "${parentId}") }`
    : gql`{ itemRepetition }`

  return function useRemoteLineItems () {
    const [line, setLine] = useState({})

    const { data } = useQuery(query, SSR ? {} : { pollInterval: 1000, nextFetchPolicy: 'cache-and-network' })

    useEffect(() => {
      const repetition = data?.itemRepetition
      if (!repetition) return setLine({})
      setLine({
        itemRepetition: {
          term: <>x 10<sup>{repetition}</sup></>,
          label: <>{repetition} {parentId ? 'repeat or self replies' : 'posts'} in 10m</>,
          modifier: (cost) => cost * Math.pow(10, repetition)
        }
      })
    }, [data?.itemRepetition])

    return line
  }
}
```

`return function useRemoteLineItems ()` is a hook which is only returned if we don't already return `() => {}`. That's what I meant with "but conditionally":

>> It's a function that returns a hook
>
> Yes but conditionally

So yes, this is conditional. That the user reported it means that the bug actually manifests already, too. Not only with my changes, assuming I wouldn't remove this line: `if (!me) return () => {}` (which what this conversation here is about)

Reply:
Sorry for the detailed and  thus maybe snarky answer, I feel like we've been talking past each other for several replies already now, lol :)